### ISSUE TITLE
Tablet Ui mini player

### ISSUE THREAD

**Issue by Angelk90** [TeamNewPipe/NewPipe#6912]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.

#### What's your question(s)?

Discuss the correct ui to use in the tablet version.
The images are based on version 0.21.9

I wanted to open a discussion, but it has not yet been decided whether to enable them or not.

As you can see from the screenshots, on the tablet version of NewPipe the mini video player is impossible to use.

The mini video player takes its full length, but the video is not seen properly.

I would suggest the mini video player like the one used by youtube.

#### Additional context

Youtube:
![image](https://user-images.githubusercontent.com/20476002/129245855-65740768-f0f8-4630-a4e5-0c5a81ca0b1f.png)

NewPipe:
![image](https://user-images.githubusercontent.com/20476002/129245841-317bb3a9-20cc-4de8-9547-c7d3763ec5d8.png)

**Comment by SameenAhnaf** [comment]
Personally, I think that mini player makes no sense. It requires the same amount of data as main player. Yet, only audio is useful.

It should be replaced by "Mini popup", I think. Interface should be accessed by a single tap just like main or popup player.

Mini popup without interface

![IMG_20210813_111214](https://user-images.githubusercontent.com/79245641/129308549-1b71d64d-72d9-4c75-9376-1e360ac7acbf.jpg)

Mini popup with interface

![IMG_20210813_111253](https://user-images.githubusercontent.com/79245641/129308559-b255c367-ccf0-402b-8423-5f753ae41403.jpg)

**Comment by Angelk90** [comment]
@SameenAhnaf : That is, do you mean to replace what you see in image two, with the enabling of the video popup?

**Comment by SameenAhnaf** [comment]
@Angelk90 Yes, I'd like to completely ditch current mini player (main playback only, not for background or popup) and replace it with following interface for all devices.Notification player could be completely ditched for main playback as well.

All buttons are not necessary. Only following buttons should get the job done.
Upwards arrow, on top right corner: Enter video info
Pause, in the middle: Pause the video
X, on bottom right corner: Close the video
Swipe up and bottom gestures could be added for entering video info and closing player. But they are not highly necessary. 

Unlike YouTube mini player, only pause, close and full screen buttons are required to be shown. No need to show the buttons all the time. Let them be a part of video interface.

**Comment by Angelk90** [comment]
I'm not very sure I understand how you would like to do it, I'll give you an example of how I had in mind with the steps, so tell me what you think.

First scenario:
1) I open app
2) I am looking for a video
3) I select the video I want to play
4) The video is playing, in this case I would remove the notification player
I would activate the notification player, only in the following cases:
- When you click on the button in the background
- When switching to another featured app
- When NewPipe loses the foreground

Second scenario:
Consider that I have the video playing, first scenario step 4).
I want to search for another video, but I want to keep playing the video so I can see and hear it.
1) I do the following gesture, I hold down on the video and I bring the video from top to bottom, as the youtube app does.
Scenario I would have the following picture:

![image](https://user-images.githubusercontent.com/20476002/129329137-04ab5087-1fa7-4bdc-b831-3f4b2beac0d1.png)

I have all the information and options I need:
Play / pause button, exit button, video information video name and channel name
I don't need anything else.

2) I can search for another video, in the meantime I can continue to see and listen to the video being played.
For example, if I can't find the video I was looking for or I don't want to play it anymore, I can go back to the previous video.

For this scenario I would not use the popup this what you see in the image:

![image](https://user-images.githubusercontent.com/20476002/129330018-6e560d04-12f1-4338-bdc1-1ff5297f9d28.png)

For the following reasons:
- Popup must be enabled, not all of them do.
- Popup video player size a little too small, I can't see the video well
- If I want to stop the video I have to click on the video
- Too many controls, as you can see from the image
- If subtitles are enabled, as you can see from the image with the popup they do not work properly.

Can I ask you to give such an example with the steps to better understand how you would like to do?

**Comment by SameenAhnaf** [comment]
Let me clarify at first how notification player exactly works for now. Notification player is added so that users can manage player for multitasking. <b> At this moment, you can't control whether to have notification player. It always appears whenever a media is playing. </b> However, it's useless for main player as all controls are in the front. 

>For this scenario I would not use the popup this what you see in the image:

I agree with you. That's why, I'm suggesting to have only three buttons.

1. Full Screen: On the top right corner
2. Pause: In the middle
3. Close: On the bottom right corner

![IMG_20210813_152149](https://user-images.githubusercontent.com/79245641/129335617-69a19b4e-a9fb-470b-a2a9-9b3dca436f1c.jpg)

**Comment by Angelk90** [comment]
@SameenAhnaf : Would this image be the popup as it should be for you or the mini player?

**Comment by SameenAhnaf** [comment]
@Angelk90 My last image will be the replacement of current mini player (only for main playback, not for background and popup playback).

Also, my last image should be available for all devices including tabs, smartphones and TV.

**Comment by Angelk90** [comment]
@SameenAhnaf : 
But how come you don't like such a thing: 
![image](https://user-images.githubusercontent.com/20476002/129346873-5f08a07a-e6c4-4d1a-86f1-053d071eae09.png)

**Comment by SameenAhnaf** [comment]
Why is the bottom part always required to be shown? It's not that hard to remember at least what the video is all about. Plus, video could be slightly expanded.

Also, 'Pause' and 'X' buttons are too close and this may cause accidental touches on smartphones.

![IMG_20210813_170657](https://user-images.githubusercontent.com/79245641/129348485-285fb5ce-5a76-4d86-9477-a8e1880fd4bf.jpg)

**Comment by Angelk90** [comment]
@SameenAhnaf : I'll give you the following example for the title:

You have found a video about something that has a title that is difficult to remember, think of a foreign title that does not belong to your language for example.
You want to look for other videos with that title, but you don't remember the wrong title of the letters for example.

Having it there would allow you to facilitate everything, not having to click on the video every time to see the title.

A trivial example, but it has happened to me a few times.

For the problem of pauses and x I never understood, I did several tests.

If the mini player will be like the image you posted, however, I would put the video progress bar at the end of the image when the video is playing, as seen in the image I posted.

**Comment by SameenAhnaf** [comment]
What about this type of layout? I think, it will solve both of our problems.
![IMG_20210813_173844](https://user-images.githubusercontent.com/79245641/129352146-33e23c34-4556-44f6-8c44-f23bfeafc780.jpg)


>For the problem of pauses and x I never understood, I did several tests.

That's probably because you own a tablet. But on smartphone, accidental touch is a pain in the ass as the buttons are way too close. 

>I would put the video progress bar at the end of the image when the video is playing

Yeah, that's a useful feature.

**Comment by Angelk90** [comment]
@SameenAhnaf : Perfect for the tablet version is excellent in my opinion, then you have to try it to see the user experience.

For the smartphone version how it will be, I took a look at how it is implemented by youtube.

Youtube app smartphone vertical:
![image](https://user-images.githubusercontent.com/20476002/129354248-3318215b-68f1-4d88-9f0b-f172a329d627.png)

Youtube app smartphone horizontal:
![image](https://user-images.githubusercontent.com/20476002/129354268-863d95a8-a46a-4e74-96ba-6da702a8cec0.png)

**Comment by SameenAhnaf** [comment]
Isn't the mini video too small in YouTube app for smartphones? What about this layout instead?

'X' on the top right corner of video.
'Pause' on the right of bottom line strip.
Single tap on title bar or swipe up the main player: Enter video info

![IMG_20210813_180637](https://user-images.githubusercontent.com/79245641/129354889-de55425b-86f3-4b58-bbd0-524ef9809b0f.jpg)

**Comment by Angelk90** [comment]
@SameenAhnaf : 

> Il mini video non è troppo piccolo nell'app YouTube per smartphone?


So the youtube developers have implemented, yes it is very small.


> 
> "X" nell'angolo in alto a destra del video.
> 'Pausa' a destra della striscia di riga inferiore.
> Tocca una volta la barra del titolo o scorri verso l'alto sul player principale: inserisci le informazioni sul video
> 
> ![IMG_20210813_180637](https://user-images.githubusercontent.com/79245641/129354889-de55425b-86f3-4b58-bbd0-524ef9809b0f.jpg)

So it would be great, another solution could also be to put the bottom row like this.
Play / pause - Title bar / channel - exit button


Will you take care of the development of this?
What do others think?

**Comment by SameenAhnaf** [comment]
>Will you take care of the development of this?

Unfortunately, I can't. I'm not a developer. Hopefully, it should catch the attention of interested devs as the issue is open.

**Comment by litetex** [comment]
As there are still no discussions here enabled (and I don't have rights to do so) I added the "discussion" label here.
Have fun 😄
